### DISCUSSION TITLE
i3xrocks icon colors

### DISCUSSION THREAD

**Discussion by ToS0** [regolith-linux/regolith-desktop#1016] [Q&A] [answered]
In some i3wm demos I saw the icons of the i3blocks had all different colors in order to faster recognize the distinct values in the bar.

Is this possible to achieve in Regolith as well?

I didn't find the right approach in the i3xrocks configs and Regolith looks.

Thank you!

**Comment by kgilmer** [comment, marked as answer]
Colors for X11 (i3blocks) are configured via Xresources 

```
$ cat lascaille/i3xrocks | grep color
i3xrocks.label.color:       color_base0
i3xrocks.value.color:       color_base00
i3xrocks.critical.color:    color_red
i3xrocks.error.color:       color_orange
i3xrocks.warning:           color_yellow
i3xrocks.nominal:           color_base0
$ pwd
/usr/share/regolith-look
```

So the color values are defined by the Look you have set.  If you want to change the color of a specific block, you'll need to edit the block file.   For example `/usr/share/i3xrocks/scripts/time` has lines like the following:

```
LABEL_COLOR=${label_color:-$(xrescat i3xrocks.label.color "#7B8394")}
VALUE_COLOR=${color:-$(xrescat i3xrocks.value.color "#D8DEE9")}
```

You would need to modify these files to specify a unique colors.  The block scripts are currently not setup to allow a per-block color spec unfortunately.

**Comment by ToS0** [comment]
Thanks a lot, this was what I missed!

On Ubuntu the paths are a bit different (e.g. `/usr/share/i3xrocks`), and I had to do both, define a new color in Xresources as well as to use it in the block script, but everything works great.

I found even better icons in the Ayu look (`/etc/regolith/styles/ayu/typeface`) and used them as well. Works and looks great!
